File: src/main/java/com/beat/global/external/notification/slack/vo/block/HeaderBlock.java
package com.beat.global.external.notification.slack.vo.block;

import static com.beat.global.external.notification.slack.vo.SlackConstant.BLOCK_TYPE_HEADER;

import com.beat.global.external.notification.slack.vo.text.PlainText;
import com.beat.global.external.notification.slack.vo.text.Text;

public record HeaderBlock(
	String type,
	Text text
) implements Block {

	public static HeaderBlock newInstance(String text) {
		return new HeaderBlock(BLOCK_TYPE_HEADER, PlainText.newInstance(text));
	}
}

File: infrastructure/src/main/kotlin/com/beat/infrastructure/external/notification/slack/vo/block/HeaderBlock.kt
package com.beat.infrastructure.external.notification.slack.vo.block

import com.beat.infrastructure.external.notification.slack.vo.SlackConstant.BLOCK_TYPE_HEADER
import com.beat.infrastructure.external.notification.slack.vo.text.PlainText
import com.beat.infrastructure.external.notification.slack.vo.text.Text

internal data class HeaderBlock(
    override val type: String,
    val text: Text,
) : Block {
    companion object {
        fun newInstance(text: String): HeaderBlock = HeaderBlock(BLOCK_TYPE_HEADER, PlainText.newInstance(text))
    }
}
